pkg/serving/config_changes.go

```diff
@@ -31,14 +32,20 @@ var UserImageAnnotationKey = "client.knative.dev/user-image"
 
 // UpdateEnvVars gives the configuration all the env var values listed in the given map of
 // vars.  Does not touch any environment variables not mentioned, but it can add
-// new env vars and change the values of existing ones.
+// new env vars and change the values of existing ones, then sort by env key name.
 func UpdateEnvVars(template *servingv1alpha1.RevisionTemplateSpec, toUpdate map[string]string, toRemove []string) error {
 	container, err := ContainerOfRevisionTemplate(template)
 	if err != nil {
 		return err
 	}
-	envVars := updateEnvVarsFromMap(container.Env, toUpdate)
-	container.Env = removeEnvVars(envVars, toRemove)
+	updated := updateEnvVars(container.Env, toUpdate)
+	removed := removeEnvVars(updated, toRemove)
```

Comment:
removeEnvVars returning the final set of env vars to be updated, isnt using 'removed' var to represent that seems odd?

I'd use `envVars` on the left side of `=` for both `updateEnvVars` and `removeEnvVars`.